webpack/ForemanTasks/Components/TasksTable/TasksTable.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from 'foremanReact/components/common/table';
+import { STATUS } from 'foremanReact/constants';
+import MessageBox from 'foremanReact/components/common/MessageBox';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { getURIQuery } from 'foremanReact/common/helpers';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import createTasksTableSchema from './TasksTableSchema';
+import { updateURlQuery } from './TasksTableHelpers';
+import './TasksTable.scss';
+
+const TasksTable = ({
+  getTableItems,
+  error,
+  status,
+  results,
+  loading,
+  query,
+  itemCount,
+  pagination,
+}) => {
+  useEffect(() => {
+    getTableItems();
+  }, [query.state, query.result, query.mode, query.time]);
+  const onPageChange = paginationArgs => {
+    updateURlQuery({
+      page: paginationArgs.page,
+      per_page: paginationArgs.perPage,
+    });
+    getTableItems();
+  };
+
+  if (status === STATUS.ERROR) {
+    return (
+      <MessageBox
+        key="tasks-table-error"
+        icontype="error-circle-o"
+        msg={__(`Could not receive data: ${error && error.message}`)}
+      />
+    );
+  }
+
+  if (loading) {
+    return <Spinner size="lg" loading />;
```

Comment:
Does showing a spinner gives a good user experience here?
I am wondering if showing blank would give better experience.
TBH spinners are nice but when the data loads fast it gives an annoying flickering effect.

Another option to use something like delayed-spinner that only shows a spinner when it takes more than 300ms to load the data.

Another option which is my favorite is using `react-loading-skeleton`, we added it to the vendor 0.2.0 so it's available already.
See: https://github.com/dvtng/react-loading-skeleton

If you decide to go with the `skelaton`, I would be OK merging it with the spinner and add this a `TableSkelaton` to patternfly-react and then use it when it's available.

Reply:
This logic is removed in the [next PR](https://github.com/theforeman/foreman-tasks/pull/445/commits/b7b9e2465e3f9d630fdd9094ea2e96cab56dfcc6) and instead the spinner is shown in the buttons row, and the table stays with the old data until its loaded, and when there is no data the table isn't shown, can you take a look?

Reply:
Thinking UX, I would suggest using an image of blurred table instead of a spinner
so the table won't jump and switch between the spinner to table.
I would also add debounce so the transitions wouldn't be too fast,
there are times when the table appears and then the spinner shows up again, but this can be in the next PR.